xdem/spstats.py

```diff
@@ -0,0 +1,507 @@
+                list_df.append(df)
+        else:
+            print('Using '+str(nproc)+ 'cores...')
+            list_dh_sub = []
+            list_coords_sub = []
+            for i in range(nrun):
+                dh_sub, coords_sub = random_subset(dh, coords, nsamp)
+                list_dh_sub.append(dh_sub)
+                list_coords_sub.append(coords_sub)
+
+            pool = mp.Pool(nproc, maxtasksperchild=1)
+            argsin = [{'dh': list_dh_sub[i], 'coords_sub': list_coords_sub[i],'i':i,'max_i':nrun} for i in range(nrun)]
+            list_df = pool.map(partial(wrapper_get_empirical_variogram,**kwargs), argsin, chunksize=1)
+            pool.close()
+            pool.join()
+
+            list_df_nb = []
+            for i in range(10):
+                df_nb = list_df[i]
+                df_nb['run'] = i
+                list_df_nb.append(df_nb)
+            df = pd.concat(list_df_nb)
+
+    return df
+
+
+def exact_neff_sphsum_circular(area: float,crange1: float,psill1: float,crange2: float,psill2: float) -> float:
+    """
+    Number of effective samples derived from exact integration of sum of spherical variogram models over a circular area.
+    The number of effective samples serves to convert between standard deviation/partial sills and standard error
+    over the area.
+    If SE is the standard error, SD the standard deviation and N_eff the number of effective samples, we have:
+    SE = SD / sqrt(N_eff) => N_eff = SD^2 / SE^2 => N_eff = (PS1 + PS2)/SE^2 where PS1 and PS2 are the partial sills
+    estimated from the variogram models, and SE is estimated by integrating the variogram models with parameters PS1/PS2
+    and R1/R2 where R1/R2 are the correlation ranges.
+    Source: Rolstad et al. (2009), appendix: http://dx.doi.org/10.3189/002214309789470950
+
+    :param area: circular area
+    :param crange1: range of short-range variogram model
+    :param psill1: partial sill of short-range variogram model
+    :param crange2: range of long-range variogram model
+    :param psill2: partial sill of long-range variogram model
+
+    :return: number of effective samples
+    """
+
+    #short range variogram
+    c1 = psill1 # partial sill
+    a1 = crange1  # short correlation range
+
+    #long range variogram
+    c1_2 = psill2
+    a1_2 = crange2 # long correlation range
+
+    h_equiv = np.sqrt(area / np.pi)
+
+    #hypothesis of a circular shape to integrate variogram model
+    if h_equiv > a1_2:
+        std_err = np.sqrt(c1 * a1 ** 2 / (5 * h_equiv ** 2) + c1_2 * a1_2 ** 2 / (5 * h_equiv ** 2))
+    elif (h_equiv < a1_2) and (h_equiv > a1):
+        std_err = np.sqrt(c1 * a1 ** 2 / (5 * h_equiv ** 2) + c1_2 * (1-h_equiv / a1_2+1 / 5 * (h_equiv / a1_2) ** 3))
+    else:
+        std_err = np.sqrt(c1 * (1-h_equiv / a1+1 / 5 * (h_equiv / a1) ** 3) + c1_2 * (1-h_equiv / a1_2+1 / 5 * (h_equiv / a1_2) ** 3))
+
+    return (psill1 + psill2)/std_err**2
+
+def neff_circ(area: float,list_vgm: list) -> float:
```

Comment:
Maybe `list` could be more specific? E.g. `list[Union[float, int]]`. Or wait, what exactly is the content of this list??

Reply:
Thanks! I couldn't remember the syntax to show the details of what should be inside the list